pinot-controller/src/test/java/org/apache/pinot/controller/api/resources/PinotSegmentUploadDownloadRestletResourceTest.java

```diff
@@ -249,4 +267,58 @@ public void testValidateMultiPartForBatchSegmentUpload() {
     // validate – should not throw exception
     PinotSegmentUploadDownloadRestletResource.validateMultiPartForBatchSegmentUpload(bodyParts);
   }
+
+  @Test
+  public void testUploadSegmentWithMissingTmpDir()
```

Comment:
I don't see this test adding anymore value than the one created in `ControllerFilePathProviderTest `, maybe you can add an integration test in `SegmentUploadIntegrationTest` to validate the segment upload flow?

Reply:
name of this test method was wrong have corrected it. This method was the one from where we received a runtime exception. Hence covering this method specifically.